internal/repository/memory/counter.go

```diff
@@ -50,6 +50,18 @@ func (m *MetricCounterMemoryRepository[T]) All() (map[string]model.Counter, erro
 	return result, nil
 }
 
+func (m *MetricCounterMemoryRepository[T]) AllAsIs() (map[string][]model.Counter, error) {
+	m.mx.RLock()
+	defer m.mx.RUnlock()
+	
+	result := make(map[string][]model.Counter)
```

Comment:
```suggestion
	result := make(map[string][]model.Counter, len(m.metrics))
```
При создании мапы также можно преаллоцировать память.